### PR TITLE
Posfix async

### PR DESCRIPTION
**Pull request by GrayJack**
Tests included :3

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -65,6 +66,7 @@ module.exports = grammar({
     [$.scoped_identifier, $.scoped_type_identifier],
     [$.parameters, $._pattern],
     [$.parameters, $.tuple_struct_pattern],
+    [$.await_expression, $.field_expression],
```
Comment (maxbrunsfeld) [inline grammar.js]: I would not expect an LR(1) conflict to arise because of await expressions. What conflict error message do you get if you remove this?

File: grammar.js [hunk 2]
```diff
@@ -1297,7 +1307,8 @@ module.exports = grammar({
 
     _reserved_identifier: $ => alias(choice(
       'default',
-      'union'
+      'union',
+      'await',
```
Comment (maxbrunsfeld) [inline grammar.js]: Possible related to the conflict above. Why did you add this to `reserved_identifier`?

File: corpus/async.txt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+futures.await;
+
+futures.await?;
+
+futures.await?.await?;
+
+futures.await?.function().await?;
+
+
```
Comment (maxbrunsfeld) [inline corpus/async.txt]: Nitpick: could you 🔥 this second blank line? Also, I think you could kill the blank lines between each statement.

File: grammar.js (hunk 2 above)
Comment (GrayJack) [reply grammar.js]: Due to the nature of the await postfix syntax, it only appears after a expression. so it's use anywhere eslse should be of a identifier

File: corpus/async.txt (hunk 3 above)
Comment (GrayJack) [reply corpus/async.txt]: Sure

File: grammar.js (hunk 2 above)
Comment (maxbrunsfeld) [reply grammar.js]: Yeah, I think you can remove this for that reason. This is only needed for tokens that can occur in expression or parameter position. I'm hoping that once you do this, you can remove the conflict that you added.

File: grammar.js (hunk 1 above)
Comment (GrayJack) [reply grammar.js]: None, I'll remove on the next commit

File: grammar.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const PREC = {
+  await:15,
```
Comment (maxbrunsfeld) [inline grammar.js]: Also, I think you can use `PREC.field` (since `x.await` probably has the same precedence as `x.y`) rather than adding a new precedence constant.

**Comment by maxbrunsfeld** [comment]
Thanks!
